lsp-mode.el

```diff
@@ -5693,7 +5693,8 @@ EXTRA is a plist of extra parameters.
 REFERENCES? t when METHOD returns references."
   (let ((loc (lsp-request method
                           (append (lsp--text-document-position-params) extra))))
-    (if (or (null loc) (hash-table-empty-p loc))
+    (if (or (seq-empty-p loc)
+            (and (hash-table-p loc) (hash-table-empty-p loc)))
```

Comment:
How about this? I think this is simple enough for now.

> The problem here is seq-empty-p don't know how to deal with hash table, how about define an cl-defmethod for seq-empty-p with hash table specializer? Or you can define new lsp-seq-empty? that works.

I don't know if we need to specify a `cl-defmethod` here. Do we use it somewhere else besides this place? 😕 

> Btw, the loc here can be either list, vector or object of plist or hash table depends on if plist feature is enabled or not

Got it. I have reverted back to `seq-empty-p`; and by pass in `nil` to function `seq-empty-p` also returns true, which means it handle the same way as the function `null`.

Ping. @kiennq and @shajra

Reply:
> I don't know if we need to specify a `cl-defmethod` here. Do we use it somewhere else besides this place?

I think we tend to have wrapper for that (like `lsp-seq-first`), but I think a simple `cl-defmethod` would also solve the problem cleanly
Some thing like
``` elisp
(cl-defmethod seq-empty-p ((ht hash-table))
  (ht-empty? ht))
```
Based on the spec https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_definition, the response can be

> Response:
> - result: Location | Location[] | LocationLink[] | null
> - partial result: Location[] | LocationLink[]
> - error: code and message set in case an exception happens during the definition request

thus, the hash table here is not relevant and depends on `lsp-use-plists` is enabled or not. For better abstraction, I think we shouldn't check the hash table type here, but instead using either wrapper or an `cl-defmethod` for `seq-empty-p`

Reply:
Got it! I see want you mean here. 

Everything should be working now! Thanks!